src/luarocks/cfg.lua

```diff
@@ -159,11 +167,18 @@ do
    end
 end
 
-local env_for_config_file = {
+local env_for_config_file
+env_for_config_file = {
    home = cfg.home,
    lua_version = cfg.lua_version,
    platform = util.make_shallow_copy(detected),
    processor = proc,
+   os_getenv = os.getenv, 
+   __dump_env = function()
```

Comment:
I'm not a fan of using "__" as a prefix (clashes with Lua's standard stuff in other places).